service/matching/matching_engine.go

```diff
@@ -1987,6 +2119,114 @@ func (e *matchingEngineImpl) pollTask(
 	return pm.PollTask(ctx, pollMetadata)
 }
 
+// isValidName checks if each character is a letter/number in the input string
+func isValidName(input string) bool {
+	for _, char := range input {
+		if !unicode.IsLetter(char) && !unicode.IsDigit(char) {
+			return false
+		}
+	}
+	return true
+}
+
+func (e *matchingEngineImpl) validateDeploymentWfParams(deploymentName string, buildID string) error {
+	// Length check
+	if deploymentName == "" || buildID == "" {
+		return serviceerror.NewInvalidArgument("DeploymentName/BuildID cannot be empty")
+	}
+
+	// Prefix check
```

Comment:
what's wrong if a deployment name or build id has one of these prefixes? it may be confusing but does anything break?

Reply:
nothing breaks (as of now) since we separate the prefixes from the name with a delimeter (`:`). However, I do wonder if we want users to be allowed to keep their deployment name as a constant we use internally.

now that you say it though, I don't see much problem in it and shall remove it - thanks for making me think again on this

Reply:
Closing this because changes have been made to reflect this.